sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/StartSpanOptions.java

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.util.tracing;
+
+import com.azure.core.annotation.Fluent;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents span options that are available before span starts and describe it.
+ */
+@Fluent
+public class StartSpanOptions {
+
+    /**
+     * Type of Span. Only CLIENT and INTERNAL kinds are supported.
+     */
+    public enum Kind {
+        /** Default value. Indicates that the span is used internally. */
+        INTERNAL,
+
+        /**
+         * Indicates that the span covers the client-side wrapper around an RPC or other remote request.
+         */
+        CLIENT,
+    }
+
+    private boolean makeCurrent;
+    private final Kind spanKind;
+    private Map<String, Object> attributes;
+
+    /**
+     * Describes span with given name and INTERNAL kind
+     */
+    public StartSpanOptions() {
+        this(Kind.INTERNAL);
+    }
+
+    /**
+     * Describes span with given name and kind
+     *
+     * @param kind The kind of the span to be created, only INTERNAL and CLIENT are supported.
+     */
+    public StartSpanOptions(Kind kind) {
+        Objects.requireNonNull(kind, "'kind' cannot be null.");
+        this.spanKind = kind;
+        this.attributes = null;
+        this.makeCurrent = false;
+    }
+
+    /**
+     * Sets flag that controls if span should be made current after it's started.
+     * @param makeCurrent flag indicating if span should be made crurent after start..
+     *
+     * @return this instance for chaining.
+     */
+    public StartSpanOptions setMakeCurrent(boolean makeCurrent) {
+        this.makeCurrent = makeCurrent;
+        return this;
+    }
+
+    /**
+     * Sets attribute on span before its started. Such attributes may affect sampling decision.
+     *
+     * @param key attribute key.
+     * @param value attribute value. Note that underlying tracer implementations limit supported value types to
+     * String, int, double, boolean, long and arrays of them.
+     *
+     * @return this instance for chaining.
+     */
+    public StartSpanOptions setAttribute(String key, Object value) {
```

Comment:
Do we expect the users to put a single attribute at a time?
In that case, should this be updated to 
```java
public StartSpanOptions putAttribute(String key, Object value) {
```

In general, I think we could update this method to
```java
public StartSpanOptions setAttribute(Map<String, String> attributeData)
````

Reply:
I believe setting attributes one by one is more friendly and performant way (we don't ask them to create another map and copy it over). In some cases, passing a map may be preferred, but we can have two apis. 

Also if you look at the Tracer, it only has `setAttribute(key, value)` there